src/utils/resolveL1OpStackContractAddress.test.ts

```diff
@@ -0,0 +1,45 @@
+import type { Chain } from 'viem'
+import { expect, test } from 'vitest'
+import { publicClient } from '../_test/utils.js'
+import { base } from '../chains/base.js'
+import { OpStackL1Contract } from '../types/opStackContracts.js'
+import { resolveL1OpStackContractAddress } from './resolveL1OpStackContractAddress.js'
+
+test('resolves Base OptimismPortal contract address', async () => {
+  const resolvedPortalAddress = await resolveL1OpStackContractAddress(
+    {
+      l2Chain: base,
+      chain: publicClient.chain,
+      contract: OpStackL1Contract.OptimismPortal,
+    },
+  )
+
+  expect(resolvedPortalAddress).toEqual('0x49048044D57e1C92A77f79988d21Fa8fAF74E97e')
+})
+
+test('raises error if chain undefined', async () => {
+  await expect(async () =>
+    resolveL1OpStackContractAddress(
+      {
```

Comment:
```suggestion
      // @ts-expect-error
      {
```
Looks like it's working correctly. It should be throwing an error if we are passing in the wrong arguments. When we expect an error in a test we use // @ts-expect-error which will throw an error if an error doesn't exist